hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java

```diff
@@ -84,13 +84,16 @@
    * milliseconds
    * @param dtRemoverScanInterval how often the tokens are scanned for expired
    * tokens in milliseconds
+   * @param certClient certificate client to SCM CA
    */
   public OzoneDelegationTokenSecretManager(OzoneConfiguration conf,
       long tokenMaxLifetime, long tokenRenewInterval,
       long dtRemoverScanInterval, Text service,
-      S3SecretManager s3SecretManager) throws IOException {
+      S3SecretManager s3SecretManager, CertificateClient certClient)
+      throws IOException {
     super(new SecurityConfig(conf), tokenMaxLifetime, tokenRenewInterval,
         service, LOG);
+    setCertClient(certClient);
```

Comment:
I understood why it is done like this by adding a new method setCertClient, as the OzoneSecretManager is base class for OzoneBlockTokenSecretManager also. Does it look clean if we pass the certClient to superclass constructor?
If it is too much change I am fine with the current approach too.

Reply:
Thanks for the review, @bharatviswa504. My first version of this patch is like what you mentioned above. But it introduce many unnecessary changes to another subclass for OzoneBlockTokenSecretManager, which does not need a certClient at constructor. It also affect many tests that use a special test flag to avoid talking to SCM for kerberos based test.

The current patch makes it optional by introduce a set method for the constructor of OzoneDelegationTokenSecret manager to use only.  This way, the change is minimized.